pkgs/stdenv/generic/setup.sh

```diff
@@ -727,17 +742,23 @@ configurePhase() {
 buildPhase() {
     runHook preBuild
 
-    if [ -z "$makeFlags" ] && ! [ -n "$makefile" -o -e "Makefile" -o -e "makefile" -o -e "GNUmakefile" ]; then
+    if [ -z "$makeFlags" ] && ! [[ -n "$makefile" || -e "Makefile" || -e "makefile" || -e "GNUmakefile" ]]; then
```

Comment:
Can't this be written as a single `[[...]]` test? Something like
```
if [[ -z $makeFlags && ! ( -n $makefile || ... ) ]]; then
```
Note that quotes are not necessary inside `[[...]]`.